python/delegates/view_item_delegate.py

```diff
@@ -1603,116 +1603,116 @@ def _draw_icon_badges(self, painter, option, bounding_rect, index):
         if not badge_data:
             return
 
-        if isinstance(badge_data, QtGui.QPixmap):
-            # Default to bottom right if only a pixmap given.
+        if not isinstance(badge_data, dict):
+            # Default to bottom right if only a pixmap given for badge data.
             badge_data = {"bottom-right": badge_data}
 
-        if isinstance(badge_data, dict):
-            for position, pixmap_data in badge_data.items():
-                if not pixmap_data:
-                    continue
+        for position, pixmap_data in badge_data.items():
+            if not pixmap_data:
+                continue
 
-                # The pixmap to display for the icon badge
-                pixmap = None
-                # Inset will display the badge inside the bounding rect, otherwise the badge
-                # will be display centered over the corner corresponding to the badge position.
-                inset = True
-                if isinstance(pixmap_data, dict):
-                    pixmap = pixmap_data.get("pixmap", None)
-                    inset = pixmap_data.get("inset", True)
-                else:
-                    pixmap = pixmap_data
+            # The pixmap to display for the icon badge
+            pixmap = None
+            # Inset will display the badge inside the bounding rect, otherwise the badge
+            # will be display centered over the corner corresponding to the badge position.
+            inset = True
+            if isinstance(pixmap_data, dict):
+                pixmap = pixmap_data.get("pixmap", None)
+                inset = pixmap_data.get("inset", True)
+            else:
+                pixmap = pixmap_data
 
-                if isinstance(pixmap, QtGui.QIcon):
-                    pixmap = self._convert_icon_to_pixmap(pixmap)
-                elif isinstance(pixmap, six.string_types):
-                    pixmap = QtGui.QPixmap(pixmap)
+            if hasattr(pixmap, "pixmap"):
+                pixmap = self._convert_icon_to_pixmap(pixmap)
+            elif isinstance(pixmap, six.string_types):
+                pixmap = QtGui.QPixmap(pixmap)
 
-                if not pixmap or not isinstance(pixmap, QtGui.QPixmap):
-                    # Skip, invalid pixmap data.
-                    continue
+            # if not pixmap or not isinstance(pixmap, QtGui.QPixmap):
+            if not pixmap:
```

Comment:
Hum, looks like something is not finished here with the commented line above.

Reply:
Oops, forgot to remove that commented line. I am removing the sanity check that the pixmap is a valid QPixmap, similar to below about the QIcon check, it may actually be better to put the onus on the caller to pass the correct object type and if not, get an error message instead of silently passing on.